src/naive/datetime/serde.rs

```diff
@@ -904,62 +2266,102 @@ pub mod ts_seconds {
         type Value = NaiveDateTime;
 
         fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-            formatter.write_str("a unix timestamp")
+            formatter.write_str("a unix timestamp string in seconds")
         }
 
-        fn visit_i64<E>(self, value: i64) -> Result<Self::Value, E>
+        /// Deserialize a string timestamp in seconds since the epoch
+        fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
         where
             E: de::Error,
         {
-            NaiveDateTime::from_timestamp_opt(value, 0)
-                .ok_or_else(|| E::custom(ne_timestamp(value)))
+            match value.parse::<i64>() {
+                Ok(value) => NaiveDateTime::from_timestamp_opt(value, 0)
+                    .ok_or_else(|| E::custom(ne_timestamp(value))),
+                Err(_) => Err(de::Error::invalid_value(Unexpected::Str(value), &self)),
+            }
         }
+    }
 
-        fn visit_u64<E>(self, value: u64) -> Result<Self::Value, E>
-        where
-            E: de::Error,
-        {
-            NaiveDateTime::from_timestamp_opt(value as i64, 0)
-                .ok_or_else(|| E::custom(ne_timestamp(value)))
+    #[cfg(all(test, feature = "serde"))]
+    mod tests {
+        use crate::naive::serde::ts_seconds_string;
+        use crate::naive::{NaiveDate, NaiveDateTime};
+        use serde_derive::{Deserialize, Serialize};
+
+        #[derive(Deserialize, Serialize)]
+        struct S {
+            #[serde(with = "ts_seconds_string")]
+            time: NaiveDateTime,
+        }
+
+        #[test]
+        fn test_serialize_naive_ts_seconds_string() -> Result<(), serde_json::Error> {
+            let my_s = S {
+                time: NaiveDate::from_ymd_opt(2015, 5, 15).unwrap().and_hms_opt(10, 0, 0).unwrap(),
+            };
+
+            let as_string = serde_json::to_string(&my_s)?;
+
+            assert_eq!(as_string, r#"{"time":"1431684000"}"#);
+
+            Ok(())
+        }
+
+        #[test]
+        fn test_deserialize_naive_ts_seconds_string() -> Result<(), serde_json::Error> {
+            const JSON: &'static str = r#"{"time":"1431684000"}"#;
+
+            let my_s: S = serde_json::from_str(JSON)?;
+
+            assert_eq!(my_s.time.timestamp(), 1431684000);
+
+            Ok(())
```

Comment:
Same question about testing maximum and minimum boundaries, and values immediately outside those boundaries.